src/components/Map/MapContainer.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {
+  createRef,
+  useState,
+} from 'react';
+import { Map, TileLayer, Marker, Popup, Circle } from 'react-leaflet';
+import { LatLngExpression, LatLngBoundsExpression } from 'leaflet';
+import { MapSidebar } from './MapSidebar';
+
+export interface Parameters {
+  capacity: string;
+  radius: number;
+  treeManagement: string;
+  conversion: string;
+  debtRatio: string;
+  interest: string;
+  debtTerm: string;
+  lifeOfProject: string;
+}
+
+export const MapContainer = () => {
+  const [mapState, setMapState] = useState({ lat: 38.538762, lng: -121.75305 });
+  const [showSidebar, toggleSidebar] = useState(true);
+  const [parameters, setParameters] = useState({
```

Comment:
@laholstege is there a strongly typed version of useState?  It would be great to be able to do something like `useState<Parameters>`.  Then you'd have good typing and wouldn't have to do that `updateParameters` hack

Reply:
@srkirkland yesterday I was having trouble getting it to work by just passing `setParameters` but now it looks like it's working. maybe because I'm passing the props correctly now? 🤷‍♀